test-broken-links.js

```diff
@@ -0,0 +1,73 @@
+const fs = require("fs");
+
+const WEBSITE_DIRECTORY = "https://legend.finos.org/";
+const DOC_DIRECTORY = "website/static/resource/studio/documentation/";
+
+var files = fs.readdirSync(DOC_DIRECTORY);
+
+for (let i = 0; i < files.length; i++) {
+  if (!files[i].endsWith(".json")) continue;
+  var jsonObject = JSON.parse(
+    fs.readFileSync(DOC_DIRECTORY + files[i], "utf8")
+  );
+
+  const keys = Object.keys(jsonObject.entries);
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    if (jsonObject.entries[key].url == undefined) {
+      continue;
+    }
+    lookForUrl(jsonObject.entries[key].url);
+  }
+}
+
+async function lookForUrlWithAnchor(url) {
```

Comment:
very nice, I think the best way to do this though is to verify based on the static website content we build in the previous workflow step, that way, we are sure the paths work and the anchor are accounted for properly (with the current approach, anchors with dashes might pose a problem)

Reply:
very true about the dashes 🙇